src/endpoint/mod.rs

```diff
@@ -38,6 +39,11 @@ impl Endpoint {
             Self::Tcp(_, _) => Transport::Tcp,
         }
     }
+
+    /// Creates an `Endpoint::Tcp` from a [`SocketAddr`]
+    pub fn from_tcp_sock_addr(addr: SocketAddr) -> Self {
```

Comment:
why not implementing `From<SocketAddr> for Endpoint`?

Reply:
`SocketAddr` could be either TCP or UDP, so its not clear which transport it should be without making it explicit

Reply:
Ah, I see...